packages/slate-react/src/components/editor.js

```diff
@@ -71,130 +72,59 @@ class Editor extends React.Component {
-    this.tmp.resolves = 0
-
-    // Resolve the plugins and create a stack and schema from them.
-    const plugins = this.resolvePlugins(props.plugins, props.schema)
-    const stack = Stack.create({ plugins })
-    const schema = Schema.create({ plugins })
-    this.state.schema = schema
-    this.state.stack = stack
-
-    // Run `onChange` on the passed-in value because we need to ensure that it
-    // is normalized, and queue the resulting change.
-    const change = props.value.change()
-    stack.run('onChange', change, this)
-    this.queueChange(change)
-    this.state.value = change.value
-
-    // Create a bound event handler for each event.
-    EVENT_HANDLERS.forEach(handler => {
-      this[handler] = (...args) => {
-        this.onEvent(handler, ...args)
-      }
-    })
+  tmp = {
+    updates: 0,
+    resolves: 0,
+    value: undefined,
+    change: undefined,
   }
 
-  /**
-   * When the `props` are updated, create a new `Stack` if necessary and run
-   * `onChange` to ensure the value is normalized.
-   *
-   * @param {Object} props
-   */
-
-  componentWillReceiveProps = props => {
-    let { schema, stack } = this
-
-    // Increment the updates counter as a baseline.
-    this.tmp.updates++
-
-    // If the plugins or the schema have changed, we need to re-resolve the
-    // plugins, since it will result in a new stack and new validations.
-    if (
-      props.plugins != this.props.plugins ||
-      props.schema != this.props.schema
-    ) {
-      const plugins = this.resolvePlugins(props.plugins, props.schema)
-      stack = Stack.create({ plugins })
-      schema = Schema.create({ plugins })
-      this.setState({ schema, stack })
-
-      // Increment the resolves counter.
-      this.tmp.resolves++
-
-      // If we've resolved a few times already, and it's exactly in line with
-      // the updates, then warn the user that they may be doing something wrong.
-      if (this.tmp.resolves > 5 && this.tmp.resolves == this.tmp.updates) {
-        logger.warn(
-          'A Slate <Editor> is re-resolving `props.plugins` or `props.schema` on each update, which leads to poor performance. This is often due to passing in a new `schema` or `plugins` prop with each render by declaring them inline in your render function. Do not do this!'
-        )
-      }
-    }
-
-    // Run `onChange` on the passed-in value because we need to ensure that it
-    // is normalized, and queue the resulting change.
-    const change = props.value.change()
-    stack.run('onChange', change, this)
-    this.queueChange(change)
-    this.setState({ value: change.value })
-  }
+  state = {}
 
   /**
    * When the component first mounts, flush any temporary changes,
    * and then, focus the editor if `autoFocus` is set.
    */
 
-  componentDidMount = () => {
-    this.flushChange()
-
+  componentDidMount() {
     if (this.props.autoFocus) {
       this.focus()
+    } else {
+      this.change(change => change)
     }
+
+    this.tmp.updates++
```

Comment:
Why does `updates` increment here?

Reply:
This updates calculates the number of completed life cycles.  It will be compared to tmp.resolves to decide whether to warn in componentDidUpdate that schema is re-created too many times.

It is ++ in DidMount because in the first mount, resolvePlugins will increase the tmp.resolves.